k9mail/src/main/java/com/fsck/k9/view/RecipientSelectView.java

```diff
@@ -160,6 +183,18 @@ public boolean onTouchEvent(@NonNull MotionEvent event) {
                 TokenImageSpan[] links = text.getSpans(offset, offset, RecipientTokenSpan.class);
                 if (links.length > 0) {
                     showAlternates(links[0].getToken());
+                    handler.removeCallbacks(recipientLongPressed);
+                    return true;
+                }
+            }
+        }
+        else if(text != null && event.getAction() == MotionEvent.ACTION_DOWN) {
+            int offset = getOffsetForPosition(event.getX(), event.getY());
```

Comment:
This method is getting quite long. Extract this new code block into a method called `handleTouchDownOnText()`